internal/pkg/deploy/cloudformation/stack/validate.go

```diff
@@ -392,3 +406,57 @@ func validateRootDirPath(input string) error {
 func validateContainerPath(input string) error {
 	return validatePath(input, maxDockerContainerPathLength)
 }
+
+// ValidatePubSubName validates naming is correct for topics in publishing/subscribing cases, such as naming for a
+// SNS Topic intended for a publisher.
+func validatePubSubName(name *string) error {
+	if name == nil || len(aws.StringValue(name)) == 0 {
+		return errMissingPublishTopicField
+	}
+
+	// name must contain letters, numbers, and can't use special characters besides _ and -
+	if !awsSNSTopicRegexp.MatchString(aws.StringValue(name)) {
+		return errInvalidPubSubTopicName
+	}
+
+	return nil
+}
+
+func validateWorkerNames(names []string) error {
+	for _, name := range names {
+		err := validateName(name)
+		if err != nil {
+			return fmt.Errorf("worker name `%s` is invalid: %w", name, err)
+		}
+	}
+	return nil
+}
+
+func validateName(name string) error {
+	if name == "" {
+		return errInvalidName
+	}
+	if len(name) > 255 {
+		return errNameTooLong
+	}
+	if !isCorrectFormat(name) {
+		return errNameBadFormat
+	}
+
+	return nil
+}
+
+func isCorrectFormat(s string) bool {
```

Comment:
Again, I think this is also only for the correct worker name format?

Reply:
This gives the correct format for any service name, so I renamed it to isCorrectSvcNameFormat.